dev/com.ibm.ws.ejbcontainer.mdb/src/com/ibm/ws/ejbcontainer/mdb/internal/MessageEndpointFactoryImpl.java

```diff
@@ -108,7 +114,14 @@ public void initialize(EJSContainer ejsContainer,
         // binding file, or defaulted to "[app/]module/bean" conforming to the
         // java:global rules for binding an EJB.
         activationSvcId = bmd.ivActivationSpecJndiName;
-        name = "messageEndpoint-" + activationSvcId;
+
+        //Register this instance as a PauseableComponent
+        name = beanMetaData.enterpriseBeanName;
+        BundleContext bundleContext = FrameworkUtil.getBundle(getClass()).getBundleContext();
+        ServiceRegistration<PauseableComponent> reg = bundleContext.registerService(PauseableComponent.class, this, null);
```

Comment:
You need a corresponding unregister for the service, such that it does not overlap subsequent copies of the MessageEndpointFactory that get created.  And does PausableComponent typically have any service properties?